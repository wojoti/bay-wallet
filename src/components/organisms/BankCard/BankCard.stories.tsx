import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import BankCard from './BankCard';

export default {
  title: 'Organisms/BankCard',
  component: BankCard,
} as ComponentMeta<typeof BankCard>;
const Template: ComponentStory<typeof BankCard> = args => (
  <BankCard {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  cardTheme: 'purple',
  name: 'Card name',
  value: '$21,37',
  cardNumber: '123456789',
};
