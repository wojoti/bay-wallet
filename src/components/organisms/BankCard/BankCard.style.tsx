import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  wrapper: {
    height: 180,
    marginBottom: 30,
  },
  align: {flex: 1, width: '85%'},
  spacer: {
    paddingVertical: 35,
    paddingLeft: 25,
    paddingRight: 50,
  },
  wave1: {
    position: 'absolute',
    width: 400,
    height: 250,
  },
  wave2: {
    position: 'absolute',
    width: 350,
    height: 250,
  },
  wave3: {
    position: 'absolute',
    width: 300,
    height: 200,
  },
});
export default style;
