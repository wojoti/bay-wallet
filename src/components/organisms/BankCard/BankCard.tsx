import {
  Background,
  Container,
  Header,
  Img,
  ImgType,
  Label,
} from 'components/atoms';
import {GradientArea} from 'components/molecules';
import {PropsWithChildren} from 'react';
import {cardYellow} from 'theme/colors';
import style from './BankCard.style';

export type BankCardProps = PropsWithChildren<{
  testId?: string;
  cardTheme: string;
  name: string;
  value: string;
  cardNumber: string;
}>;

const BankCard = ({
  cardTheme,
  name,
  value,
  cardNumber,
  testId,
}: BankCardProps) => {
  const modifiedCardNumber =
    cardNumber.substring(0, 3) + ' ' + cardNumber.substring(3, 6) + ' ***';
  switch (cardTheme) {
    case 'purple':
      return (
        <Container customStyle={style.wrapper} align="center" testId={testId}>
          <Container customStyle={style.align}>
            <GradientArea
              colors={['#645CA1', '#C0B8FF']}
              start={{x: 0, y: 0}}
              end={{x: 1, y: 0}}>
              <Background
                source={require('img/wavecard1.png')}
                customStyle={style.wave1}
              />
              <Container flex={1} direction="row" customStyle={style.spacer}>
                <Container flex={2} justify="flex-start">
                  <Container flex={1} justify="flex-start" align="flex-start">
                    <Label color="white" testId="bankcard-name">
                      {name}
                    </Label>
                    <Header size={28} color="white" testId="bankcard-value">
                      {value}
                    </Header>
                  </Container>
                  <Container flex={1} justify="flex-end" align="flex-start">
                    <Header
                      spacing={4}
                      size={17}
                      color={cardYellow}
                      testId="bankcard-cardnumber">
                      {modifiedCardNumber}
                    </Header>
                  </Container>
                </Container>
                <Container flex={1} justify="flex-end" align="flex-end">
                  <Img
                    src={ImgType.cardlogo1}
                    height={30}
                    width={50}
                    alt={''}
                  />
                </Container>
              </Container>
            </GradientArea>
          </Container>
        </Container>
      );
    case 'green':
      return (
        <Container customStyle={style.wrapper} align="center" testId={testId}>
          <Container customStyle={style.align}>
            <GradientArea
              colors={['#44A159', '#9CEA38']}
              start={{x: 0, y: 0}}
              end={{x: 1, y: 0}}>
              <Background
                source={require('img/wavecard2.png')}
                customStyle={style.wave2}
              />
              <Container flex={1} direction="row" customStyle={style.spacer}>
                <Container flex={2} justify="flex-start">
                  <Container flex={1} justify="flex-start" align="flex-start">
                    <Label color="white" testId="bankcard-name">
                      {name}
                    </Label>
                    <Header size={28} color="white" testId="bankcard-value">
                      {value}
                    </Header>
                  </Container>
                  <Container flex={1} justify="flex-end" align="flex-start">
                    <Header
                      spacing={4}
                      size={17}
                      color={cardYellow}
                      testId="bankcard-cardnumber">
                      {modifiedCardNumber}
                    </Header>
                  </Container>
                </Container>
                <Container flex={1} justify="flex-end" align="flex-end">
                  <Img
                    src={ImgType.cardlogo2}
                    height={42}
                    width={42}
                    alt={''}
                  />
                </Container>
              </Container>
            </GradientArea>
          </Container>
        </Container>
      );
    case 'pink':
      return (
        <Container customStyle={style.wrapper} align="center" testId={testId}>
          <Container customStyle={style.align}>
            <GradientArea
              colors={['#DA1CC7', '#FFC785']}
              start={{x: 0, y: 0}}
              end={{x: 1, y: 0}}>
              <Background
                source={require('img/wavecard3.png')}
                customStyle={style.wave3}
              />
              <Container flex={1} direction="row" customStyle={style.spacer}>
                <Container flex={2} justify="flex-start">
                  <Container flex={1} justify="flex-start" align="flex-start">
                    <Label color="white" testId="bankcard-name">
                      {name}
                    </Label>
                    <Header size={28} color="white" testId="bankcard-value">
                      {value}
                    </Header>
                  </Container>
                  <Container flex={1} justify="flex-end" align="flex-start">
                    <Header
                      spacing={4}
                      size={17}
                      color={cardYellow}
                      testId="bankcard-cardnumber">
                      {modifiedCardNumber}
                    </Header>
                  </Container>
                </Container>
                <Container flex={1} justify="flex-end" align="flex-end">
                  <Img
                    src={ImgType.cardlogo3}
                    height={37}
                    width={19.5}
                    alt={''}
                  />
                </Container>
              </Container>
            </GradientArea>
          </Container>
        </Container>
      );
    case 'darkPurple':
      return (
        <Container customStyle={style.wrapper} align="center" testId={testId}>
          <Container customStyle={style.align}>
            <GradientArea
              colors={['#3e3b53', '#7c77a2']}
              start={{x: 0, y: 0}}
              end={{x: 1, y: 0}}>
              <Background
                source={require('img/wavecard1.png')}
                customStyle={style.wave1}
              />
              <Container flex={1} direction="row" customStyle={style.spacer}>
                <Container flex={2} justify="flex-start">
                  <Container flex={1} justify="flex-start" align="flex-start">
                    <Label color="white" testId="bankcard-name">
                      {name}
                    </Label>
                    <Header size={28} color="white" testId="bankcard-value">
                      {value}
                    </Header>
                  </Container>
                  <Container flex={1} justify="flex-end" align="flex-start">
                    <Header
                      spacing={4}
                      size={17}
                      color={cardYellow}
                      testId="bankcard-cardnumber">
                      {modifiedCardNumber}
                    </Header>
                  </Container>
                </Container>
                <Container flex={1} justify="flex-end" align="flex-end">
                  <Img
                    src={ImgType.cardlogo1}
                    height={30}
                    width={50}
                    alt={''}
                  />
                </Container>
              </Container>
            </GradientArea>
          </Container>
        </Container>
      );
    default:
      return (
        <Container customStyle={style.wrapper} align="center" testId={testId}>
          <Container customStyle={style.align}>
            <GradientArea
              colors={['#3e3b53', '#7c77a2']}
              start={{x: 0, y: 0}}
              end={{x: 1, y: 0}}>
              <Background
                source={require('img/wavecard1.png')}
                customStyle={style.wave1}
              />
              <Container flex={1} direction="row" customStyle={style.spacer}>
                <Container flex={2} justify="flex-start">
                  <Container flex={1} justify="flex-start" align="flex-start">
                    <Label color="white" testId="bankcard-name">
                      {name}
                    </Label>
                    <Header size={30} color="white" testId="bankcard-value">
                      {value}
                    </Header>
                  </Container>
                  <Container flex={1} justify="flex-end" align="flex-start">
                    <Header
                      spacing={4}
                      size={17}
                      color={cardYellow}
                      testId="bankcard-cardnumber">
                      {modifiedCardNumber}
                    </Header>
                  </Container>
                </Container>
                <Container flex={1} justify="flex-end" align="flex-end">
                  <Img
                    src={ImgType.cardlogo1}
                    height={30}
                    width={50}
                    alt={''}
                  />
                </Container>
              </Container>
            </GradientArea>
          </Container>
        </Container>
      );
  }
};

export default BankCard;
