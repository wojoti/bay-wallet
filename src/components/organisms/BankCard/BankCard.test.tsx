import '@testing-library/jest-native/extend-expect';
import {render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import BankCard, {BankCardProps} from './BankCard';

const props: BankCardProps = {
  cardTheme: 'green',
  name: 'name',
  value: 'value',
  cardNumber: '123456789123',
  testId: 'test-bankcard-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<BankCard {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render BankCard', () => {
  render(<BankCard {...props} />);
  const BankCardElement = screen.getByTestId('test-bankcard-id');
  expect(BankCardElement).toBeOnTheScreen();
});

test('should render BankCard - name', () => {
  render(<BankCard {...props} />);
  const BankCardElement = screen.getByTestId('test-bankcard-id');
  expect(BankCardElement).toBeOnTheScreen();
  const labelElement = within(BankCardElement).getByTestId('bankcard-name');
  expect(labelElement).toBeOnTheScreen();
  expect(labelElement).toHaveTextContent('name');
});

test('should render BankCard - value', () => {
  render(<BankCard {...props} />);
  const BankCardElement = screen.getByTestId('test-bankcard-id');
  expect(BankCardElement).toBeOnTheScreen();
  const headerElement = within(BankCardElement).getByTestId('bankcard-value');
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveTextContent('value');
});

test('should render BankCard - modified card number', () => {
  render(<BankCard {...props} />);
  const BankCardElement = screen.getByTestId('test-bankcard-id');
  expect(BankCardElement).toBeOnTheScreen();
  const headerElement = within(BankCardElement).getByTestId(
    'bankcard-cardnumber',
  );
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveTextContent('123 456 789 ***');
});
