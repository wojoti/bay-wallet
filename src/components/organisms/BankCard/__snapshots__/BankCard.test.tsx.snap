// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "height": 180,
        "marginBottom": 30,
      },
    ]
  }
  testID="test-bankcard-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "flex": 1,
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "borderRadius": 14,
          },
        ]
      }
    >
      <BVLinearGradient
        colors={
          [
            4282687833,
            4288473656,
          ]
        }
        endPoint={
          {
            "x": 1,
            "y": 0,
          }
        }
        locations={null}
        startPoint={
          {
            "x": 0,
            "y": 0,
          }
        }
        style={
          {
            "borderRadius": 14,
            "flex": 1,
          }
        }
        testID="gradientarea-gradient"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            [
              {
                "flex": 1,
              },
              {
                "height": 250,
                "position": "absolute",
                "width": 350,
              },
            ]
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/img/wavecard2.png",
              }
            }
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": 250,
                  "width": 350,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          style={
            [
              {
                "alignItems": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "width": "100%",
              },
              {
                "paddingLeft": 25,
                "paddingRight": 50,
                "paddingVertical": 35,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "stretch",
                  "flex": 2,
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "fontFamily": "Poppins-Light",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "textAlign": "left",
                  }
                }
                testID="bankcard-name"
              >
                name
              </Text>
              <Text
                style={
                  {
                    "color": "white",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 28,
                    "letterSpacing": undefined,
                    "textAlign": "left",
                  }
                }
                testID="bankcard-value"
              >
                value
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-end",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#FFEF60",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 17,
                    "letterSpacing": 4,
                    "textAlign": "left",
                  }
                }
                testID="bankcard-cardnumber"
              >
                123 456 789 ***
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <SvgMock
              height={42}
              width={42}
            />
          </View>
        </View>
      </BVLinearGradient>
    </View>
  </View>
</View>
`;
